### PR TITLE
documentation on how to protect endpoints, applied example of milestones

### PR DESCRIPTION
**Pull request by moxious**
I recently opened an issue after having a hard time figuring out how to protect REST endpoints.  I updated the main README and would request you pull this simple example in to help out newcomers.

The existing README provides a link to milestones, but doesn't explain what they are or why you'd use them.  I've improved that and provided a code example so that people who want a quick start can do a very common thing (protect endpoints) off of the main documentation page.

### CODE REVIEW

**Comment by moxious** [comment]
Related issue which prompted me to do this:  https://github.com/dchester/epilogue/issues/127

Not sure why the travis build would fail, as I only changed README.md.

File: README.md [hunk 1]
```diff
@@ -402,7 +402,69 @@ Create a resource and CRUD actions given a Sequelize model and endpoints.  Accep
...
+To protect all endpoints, we'll use `users.all.auth`, a hook used to authorize the
+endpoint before any operation (`create`, `list`, etc).  Suppose also we have an
+express middlware function called `authorize(req, res, done)`.   This authorize
+function might for example be a passport strategy such as `passport('local')`.
+
+To authorize the endpoint, you would do this:
+
+```javascript
+users.all.auth(function (req, res, context) {
+  var resolver = Promise.defer();
```
Comment (mbroadst) [inline README.md]: http://bluebirdjs.com/docs/api/deferred-migration.html

Perhaps opt for:
```
return new Promise(function(resolve, reject) {
  authorize(req, res, function (arg) {
    if (arg) {
      // Middleware function returned an error; this means the operation
      // should not be authorized.
      res.status(401).send({message: "Unauthorized"});
      resolve(context.stop);
    } else {
      resolve(context.continue);
    }
});
```

**Comment by mbroadst** [comment]
just one comment otherwise LGTM. Thanks for the contribution!

**Comment by NatuMyers** [comment]
Would it not be userResourceName.all.auth not users.all.auth?

**Comment by mbroadst** [comment]
@NatuMyers thanks for the tip, I updated the README again for what is hopefully the last update related to this type of confusion :)

**Pull request merged**